### PR TITLE
Add ability to pass in ENV to workstation

### PR DESCRIPTION
**Pull request by damienrj**
Added a new `--envs` option to the `create` command to allow setting environment variables at runtime.

### CODE REVIEW

File: src/workstation/cli/crud.py [hunk 1]
```diff
@@ -190,6 +199,7 @@ def create(
     project: Optional[str],
     proxy: Optional[str],
     no_proxy: Optional[str],
+    envs: tuple,
```
Comment (jhamet93) [inline src/workstation/cli/crud.py]: nit: `Tuple[Tuple[str, str]]`

File: src/workstation/core.py [hunk 2]
```diff
@@ -107,6 +107,7 @@ def create_workstation(
     user: str,
     proxy: Optional[str] = None,
     no_proxy: Optional[str] = None,
+    envs: Optional[Tuple] = None,
```
Comment (jhamet93) [inline src/workstation/core.py]: nit: `Optional[Tuple[Tuple[str, str]]]`

File: src/workstation/core.py [hunk 3]
```diff
@@ -151,6 +152,14 @@ def create_workstation(
         env["no_proxy"] = no_proxy
         env["NO_PROXY"] = no_proxy
 
+    if envs:
+        user_envs = dict(envs)
+        # ensure that no duplicate keys are added to env
+        for key, value in user_envs.items():
+            if key not in env:
+                env[key] = value
```
Comment (jhamet93) [inline src/workstation/core.py:162]: nit: log the duplicate being ignored or just add validation to click to fail in this scenario

**Review by jhamet93 — APPROVED** [review]

File: src/workstation/cli/crud.py [hunk 4]
```diff
@@ -180,6 +180,15 @@ def common_options(func):  # noqa: D103
     type=str,
     metavar="<str>",
 )
+@click.option(
+    "--envs",
+    "-e",
+    "envs",
```
Comment (andrewblane) [inline src/workstation/cli/crud.py:186]: I'd have maybe expected it to be 'env' if it's passed multiple times each with one env var?

Comment (damienrj) [reply src/workstation/cli/crud.py:186]: That's fair, have debated if it should be plural or not but I think that makes more sense.

Comment (dwyatte) [reply src/workstation/cli/crud.py:186]: Agree, I would have expected something like 

`--env VAR1=foo --env VAR2=bar`

File: src/workstation/core.py [hunk 5]
```diff
@@ -107,6 +107,7 @@ def create_workstation(
     user: str,
     proxy: Optional[str] = None,
     no_proxy: Optional[str] = None,
+    envs: Optional[Tuple[Tuple[str, str]]] = None,
```
Comment (dwyatte) [inline src/workstation/core.py:110]: Looks like there is a docstring we should update too

File: src/workstation/cli/crud.py (hunk 4 above)
Comment (damienrj) [reply src/workstation/cli/crud.py:186]: Have changed it, `=` not possible with how it is setup with click so it is space

**Pull request merged**